ros2action/ros2action/verb/echo.py

```diff
@@ -0,0 +1,292 @@
+            try:
+                executor.add_node(node)
+                while executor.context.ok():
+                    executor.spin_once()
+            except KeyboardInterrupt:
+                pass
+            finally:
+                executor.remove_node(node)
+
+            if send_goal_event_sub:
+                send_goal_event_sub.destroy()
+            if cancel_goal_event_sub:
+                cancel_goal_event_sub.destroy()
+            if get_result_event_sub:
+                get_result_event_sub.destroy()
+            if feedback_sub:
+                feedback_sub.destroy()
+            if status_sub:
+                status_sub.destroy()
+
+            run_thread = False
+            if output_msg_thread.is_alive():
+                output_msg_thread.join(1)
+
+    def _send_goal_subscriber_callback(self, msg):
+        self._base_subscriber_callback(msg, ActionInterfaces.GOAL_SERVICE.value)
+
+    def _cancel_goal_subscriber_callback(self, msg):
+        self._base_subscriber_callback(msg, ActionInterfaces.CANCEL_SERVICE.value)
+
+    def _get_result_subscriber_callback(self, msg):
+        self._base_subscriber_callback(msg, ActionInterfaces.RESULT_SERVICE.value)
+
+    def _feedback_subscriber_callback(self, msg):
+        self._base_subscriber_callback(msg, ActionInterfaces.FEEDBACK_TOPIC.value)
+
+    def _status_subscriber_callback(self, msg):
+        self._base_subscriber_callback(msg, ActionInterfaces.STATUS_TOPIC.value)
+
+    def _base_subscriber_callback(self, msg, interface: str):
+        to_print = 'interface: ' + interface + '\n'
+        if self.csv:
+            to_print += message_to_csv(msg, truncate_length=self.truncate_length,
+                                       no_arr=self.no_arr, no_str=self.no_str)
+        else:
+            # The "easy" way to print out a representation here is to call message_to_yaml().
+            # However, the message contains numbers for the event type, but we want to show
+            # meaningful names to the user.  So we call message_to_ordereddict() instead,
+            # and replace the numbers with meaningful names before dumping to YAML.
+            msgdict = message_to_ordereddict(msg, truncate_length=self.truncate_length,
+                                             no_arr=self.no_arr, no_str=self.no_str)
+
+            if 'info' in msgdict:
+                info = msgdict['info']
+                if 'event_type' in info:
+                    info['event_type'] = self._event_number_to_name[info['event_type']]
+
+            to_print += yaml.dump(msgdict, allow_unicode=True, width=sys.maxsize,
+                                  default_flow_style=self.flow_style)
+
+            to_print += '---'
+        try:
+            self.output_msg_queue.put(to_print, timeout=0.5)
+        except queue.Full:
+            print('Output message is full! Please increase the queue size of output message by '
+                  '"--queue_size"')
```

Comment:
i think this generates the complication for user like `what is the appropriate queue size?` and `How many short?`. instead, maybe system allocate large enough queue size as macro with some notes?